sunpy/io/file_tools.py

```diff
@@ -169,20 +154,18 @@ def _detect_filetype(filepath):
     """
     Attempts to determine the type of data contained in a file.  This is only
     used for reading because it opens the file to check the data.
-
     Parameters
     ----------
-    filepath : `str`
+    filepath : `PurePosixPath/PosixPath`
```

Comment:
`PosixPath` and `PurePosixPath` are for use on non-windows systems only. Why is this `PosixPath`?